Remove/Change methods which are deprecated by jQuery3

timokae opened the pull request:
This pull request removes the methods [focus](https://api.jquery.com/focus/) (in [v3.3.0](https://blog.jquery.com/2018/01/19/jquery-3-3-0-a-fragrant-bouquet-of-deprecations-and-is-that-a-new-feature/)) and [delegate](https://api.jquery.com/delegate/) (in [v3.0.0](https://api.jquery.com/category/deprecated/deprecated-3.0/)) got deprecated. 

In case of `focus` the underlying [trigger](https://api.jquery.com/trigger/) method is used, which is available since v1.0.0.
For the `delegate` method, [on](https://api.jquery.com/on/) is being used instead (added in v1.7.0).

tagliala commented:
Hi @timokae, thanks for this PR

Is this change backward compatible with jQuery 1.12? I've tried in a [fiddle](https://jsfiddle.net/tagliala/8vgkxe6L/) and it looks it is

timokae commented:
I just updated my text above. It should be. I added the methods `on` (added in 1.7) and `trigger` (added in 1.0). 
Both are already available in 1.12.

tagliala commented:
@timokae could you please rebase from master?

tagliala commented:
Tested against Rails4/BS3/jQuery1.12 and Rails6/BS4/jQuery3

Thanks
